Components/Currency.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+class Shopware_Components_Paypal_Currency extends Enlight_Class
+{
+    protected $currencyId;
+
+    private $currencies = array(
+        'AUD',
+        'CAD',
+        'CZK',
+        'DKK',
+        'EUR',
+        'HKD',
+        'ILS',
+        'MXN',
+        'NZD',
+        'NOK',
+        'PHP',
+        'PLN',
+        'GBP',
+        'RUB',
+        'SGD',
+        'SEK',
+        'CHF',
+        'THB',
+        'USD'
+    );
+
+    public function __construct($config)
```

Comment:
__construct($currencyId, $currencyRepository, $shopCurrency)

don't use Shopware() in services:
 https://developers.shopware.com/blog/2015/11/11/best-practices-of-shopware-plugin-development/#shopware